cluster_test.go

```diff
@@ -66,20 +72,28 @@ func newTestCluster(t *testing.T) *testCluster {
 	root, err := ioutil.TempDir("", "sequins-cluster-")
 	require.NoError(t, err)
 
-	zkServers, _ := createTestZkCluster(t)
+	zkServers, zkCluster := createTestZkCluster(t)
+
+	// We have a specific transport to the client, so it doesn't try to reuse
+	// connections between tests
+	var testClient = &http.Client{
+		Timeout:   100 * time.Millisecond,
+		Transport: &http.Transport{},
+	}
 
 	return &testCluster{
-		T:         t,
-		binary:    binary,
-		root:      root,
-		sequinses: make([]*testSequins, 0),
-		zkServers: zkServers,
+		T:          t,
+		binary:     binary,
+		root:       root,
+		sequinses:  make([]*testSequins, 0),
+		zkServers:  zkServers,
+		zkCluster:  zkCluster,
+		testClient: testClient,
 	}
 }
 
 func (tc *testCluster) addSequins() {
-	index := len(tc.sequinses)
-	port := 19590 + index // TODO pick random high port
+	port := randomPort()
```

Comment:
I tried a few things, but didn't come up with anything good. In any case the zookeeper test cluster stuff does the same thing, so even if we fixed it here we'd have the same problem there. ¯\_(ツ)_/¯

Since the tests can be rerun easily, this seems not hugely pressing.